Tests/AssistantV1Tests/AssistantV1UnitTests.swift

```diff
@@ -0,0 +1,1823 @@
+
+        let expectation = self.expectation(description: "message")
+        assistant.message(
+            workspaceID: self.workspaceID,
+            input: input,
+            alternateIntents: true,
+            context: context,
+            entities: entities,
+            intents: intents,
+            output: output,
+            nodesVisitedDetails: true) {
+                _, _ in
+                expectation.fulfill()
+        }
+        waitForExpectations(timeout: timeout)
+    }
+
+    // MARK: - Workspaces
+
+    func testListWorkspaces() {
+        MockURLProtocol.requestHandler = { request in
+            XCTAssertEqual(request.httpMethod, "GET")
+            XCTAssertNotNil(request.allHTTPHeaderFields)
+            XCTAssertNil(request.httpBodyStream)
+
+            XCTAssertEqual(request.url?.lastPathComponent, "workspaces")
+            XCTAssertTrue(request.url?.query?.contains("version=\(currentDate)") ?? false)
+            XCTAssertTrue(request.url?.query?.contains("page_limit=10") ?? false)
+            XCTAssertTrue(request.url?.query?.contains("include_count=true") ?? false)
+            XCTAssertTrue(request.url?.query?.contains("sort=alphabetical") ?? false)
+            XCTAssertTrue(request.url?.query?.contains("cursor=mouse") ?? false)
+            XCTAssertTrue(request.url?.query?.contains("include_audit=true") ?? false)
+
+            return (HTTPURLResponse(), Data())
+        }
+
+        let expectation = self.expectation(description: "listWorkspaces")
+        assistant.listWorkspaces(
+            pageLimit: 10,
+            includeCount: true,
+            sort: "alphabetical",
+            cursor: "mouse",
+            includeAudit: true) {
+                _, _ in
+                expectation.fulfill()
+        }
+        waitForExpectations(timeout: timeout)
+    }
+
+    func testCreateWorkspace() {
+        MockURLProtocol.requestHandler = { request in
+            XCTAssertEqual(request.httpMethod, "POST")
+            XCTAssertNotNil(request.allHTTPHeaderFields)
+
+            XCTAssertEqual(request.url?.lastPathComponent, "workspaces")
+            XCTAssertTrue(request.url?.query?.contains("version=\(currentDate)") ?? false)
+
+            do {
+                let body = Data(reading: request.httpBodyStream!)
+                let decodedBody = try JSONDecoder().decode(CreateWorkspace.self, from: body)
+
+                XCTAssertNotNil(decodedBody.counterexamples)
+                XCTAssertNotNil(decodedBody.description)
+                XCTAssertNotNil(decodedBody.entities)
+                XCTAssertNotNil(decodedBody.intents)
+                XCTAssertNotNil(decodedBody.language)
+                XCTAssertNotNil(decodedBody.learningOptOut)
+                XCTAssertNotNil(decodedBody.metadata)
+                XCTAssertNotNil(decodedBody.name)
+                XCTAssertNotNil(decodedBody.systemSettings)
```

Comment:
We could make the test stronger by checking the value of these elements and not just for not nil.  It seems like that would be pretty easy to do.  Any reason we should not do that?

Reply:
Honestly, this kinda came down to laziness on my part. Writing these tests has been incredibly tedious. I tried rationalizing it, but you're right, it would be better to check the actual values to be absolutely certain that both encoding and decoding work as expected.

Reply:
I ran into a problem when trying to check the actual values of the decoded request body. None of our models conform to the `Equatable` protocol, so we can't compare the output objects to the input objects. I suppose we could do a hybrid approach, where only the native types are checked for equality. However, those are in the minority, since most body parameters are custom types, so I'm not sure that would be worth it.

Reply:
Nevermind, every struct could automatically be made equatable by just adding `Equatable` to the protocol conformances. I was able to make this update without much trouble.